access-token-management/src/AccessTokenManagement/Internal/DefaultClientCredentialsCacheKeyGenerator.cs

```diff
@@ -18,9 +20,21 @@ public ClientCredentialsCacheKey GenerateKey(
         ClientCredentialsClientName clientName,
         TokenRequestParameters? parameters = null)
     {
-        var scopePart = "s_" + parameters?.Scope;
-        var resourcePart = "r_" + parameters?.Resource;
+        var scopePart = "s_" + GetCacheKey(parameters?.Scope ?? string.Empty);
```

Comment:
@pgermishuys Is the `?? string.Empty` needed here given that `GetCacheKey` (which is really "GetMd5Hash()") takes a 'string?'. Just pass in `parameters?.Scope` as is.